plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/confluenceToMarkdown.test.ts

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { PassThrough } from 'stream';
+import { createConfluenceToMarkdownAction } from './confluenceToMarkdown';
+import { getVoidLogger } from '@backstage/backend-common';
+import { UrlReader } from '@backstage/backend-common';
+import { ConfigReader } from '@backstage/config';
+import { ScmIntegrations } from '@backstage/integration';
+import { setupRequestMockHandlers } from '@backstage/test-utils';
+import mockFs from 'mock-fs';
+import os from 'os';
+import type { ActionContext } from '@backstage/plugin-scaffolder-node';
+import { readFile, writeFile, createWriteStream } from 'fs-extra';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+
+jest.mock('fs-extra', () => ({
```

Comment:
Since jest mocks return undefined, the example you mentioned above will not work for me. I need to mock implementations on a few of the functions, or else I get errors like the one below:
```
 TypeError: Cannot read properties of undefined (reading 'on')

       95 |           `${workspace}/${mkdocsDir}docs/img/${downloadTitle}`,
       96 |         );
    >  97 |         res.body.pipe(writeStream);
          |                  ^
       98 |         await new Promise((resolve, reject) => {
       99 |           writeStream.on('finish', () => {
      100 |             resolve(`${workspace}/${mkdocsDir}docs/img/${downloadTitle}`);
```

If I try and remove the methods I mocked with just `jest.fn()`, I get the following error: 
`TypeError: _fsExtra.default.outputFile is not a function`

I was able to clean up the mocks elsewhere!